src/main/java/toothless/parser/Parser.java

```diff
@@ -0,0 +1,183 @@
+import toothless.ui.Ui;
+
+/**
+ * A class that parses, resolves and validates user input.
+ */
+public class Parser {
+    /**
+     * The main method that parses user input.
+     * @param userInput User input string to be parsed.
+     * @param tasks TaskList class that is operated on when user command is executed.
+     * @param ui Ui class used to print messages.
+     * @return String of response.
+     * @throws ToothlessException if user input is invalid.
+     */
+    public String parseInput(String userInput, TaskList tasks, Ui ui) throws ToothlessException {
+        String response = "";
+        if (userInput.equals("bye")) {
+            response = "Bye. Purr-lease chat again soon!";
+        } else if (userInput.equals("list")) {
+            response = ui.getListMessage(tasks.getTasks(),
+                    "Oops! Looks like you haven't added any tasks yet!",
+                    "Here are the tasks in your list:\n");
+        } else if (userInput.startsWith("mark ") || userInput.equals("mark")) {
+            int listIndex = validateListInput(userInput, "mark", tasks.size());
+            Task markedTask = tasks.markTask(listIndex);
+            response = ui.getMarkedTaskMessage(markedTask);
+        } else if (userInput.startsWith("unmark ") || userInput.equals("unmark")) {
+            int listIndex = validateListInput(userInput, "unmark", tasks.size());
+            Task unmarkedTask = tasks.unmarkTask(listIndex);
+            response = ui.getUnmarkedTaskMessage(unmarkedTask);
+        } else if (userInput.startsWith("todo ") || userInput.equals("todo")) {
+            String taskDescription = validateToDoInput(userInput);
+            Task newTask = tasks.addToDoToList(taskDescription);
+            response = ui.getNewTaskMessage(newTask, tasks.size());
+        } else if (userInput.startsWith("deadline ") || userInput.equals("deadline")) {
+            String[] deadlineAttributes = validateDeadlineInput(userInput);
+            Task newTask = tasks.addDeadlineToList(deadlineAttributes[0], deadlineAttributes[1]);
+            response = ui.getNewTaskMessage(newTask, tasks.size());
+        } else if (userInput.startsWith("event ") || userInput.equals("event")) {
+            String[] eventAttributes = validateEventInput(userInput);
+            Task newTask = tasks.addEventToList(eventAttributes[0], eventAttributes[1], eventAttributes[2]);
+            response = ui.getNewTaskMessage(newTask, tasks.size());
+        } else if (userInput.startsWith("delete ") || userInput.equals("delete")) {
+            int listIndex = validateListInput(userInput, "delete", tasks.size());
+            Task deletedTask = tasks.deleteTask(listIndex);
+            response = ui.getDeletedTaskMessage(deletedTask, tasks.size());
+        } else if (userInput.startsWith("find ") || userInput.equals("find")) {
+            String keyword = validateFindInput(userInput);
+            ArrayList<Task> keywordTasks = tasks.findKeyword(keyword);
```

Comment:
Perhaps this would be better written as a case statement, or with added line breaks for readability?